Source/engine/animationinfo.cpp

```diff
@@ -172,11 +173,12 @@ void AnimationInfo::ChangeAnimationData(std::optional<CelSprite> celSprite, int
 	if (numberOfFrames != NumberOfFrames || ticksPerFrame != TicksPerFrame) {
 		// Ensure that the CurrentFrame is still valid and that we disable ADL cause the calculcated values (for example TickModifier) could be wrong
 		if (numberOfFrames >= 1)
-			CurrentFrame = clamp(CurrentFrame, 0, numberOfFrames - 1);
+			CurrentFrame = clamp(CurrentFrame, 0, (int)lastFrame);

```

Comment:
Use `static_cast`, not c-style casts. Was this in response to a warning? unit8_t to int promotion is one of the guaranteed good conversions so shouldn't need a cast...

Reply:
Ye, I got rid finally of `lastFrame` at the end, if you look at `Changed Files` for review - which I recommend.

Reply:
It turned out, it would be used in really tiny number after I introduced better helpers.